dds/DCPS/XTypes/DynamicData.h

```diff
@@ -0,0 +1,277 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_XTYPES_DYNAMIC_DATA_H
+#define OPENDDS_DCPS_XTYPES_DYNAMIC_DATA_H
+
+#include "TypeObject.h"
+#include "DynamicType.h"
+
+#include <dds/DCPS/Serializer.h>
+#include <dds/DCPS/PoolAllocator.h>
+#include <dds/CorbaSeq/LongSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/ULongSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/Int8SeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/UInt8SeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/ShortSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/UShortSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/LongLongSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/ULongLongSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/FloatSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/DoubleSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/LongDoubleSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/CharSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/WCharSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/OctetSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/BooleanSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/StringSeqTypeSupportImpl.h>
+#include <dds/CorbaSeq/WStringSeqTypeSupportImpl.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace XTypes {
+
+class OpenDDS_Dcps_Export DynamicData {
+public:
+  DynamicData();
+
+  /// This creates a duplicated ACE_Message_Block chain from the provided chain.
+  /// The duplicated chain is released when the object is destroyed. Caller is
+  /// responsible for the release of the input message block chain.
+  DynamicData(ACE_Message_Block* chain,
+              const DCPS::Encoding& encoding,
+              const DynamicType_rch& type);
+  DynamicData(DCPS::Serializer& ser, const DynamicType_rch& type);
+  DynamicData& operator=(const DynamicData& other);
+
+  ~DynamicData();
+
+  DDS::ReturnCode_t get_descriptor(MemberDescriptor& value, MemberId id) const;
+  DDS::ReturnCode_t set_descriptor(MemberId id, const MemberDescriptor& value);
+
+  MemberId get_member_id_by_name(DCPS::String name) const;
+  MemberId get_member_id_at_index(ACE_CDR::ULong index);
+  ACE_CDR::ULong get_item_count();
```

Comment:
These two both look like they should be const members.

Reply:
For the same reason as the get_*_value and get_*_values functions, these two examine the underlying data to determine the number of items and compute the id. They modify the Serializer data member and can't be const.